### PR TITLE
Qt: advise users not to switch wallets when opening a BIP70 URI.

### PR DESCRIPTION
**Pull request by jameshilliard**
It would probably be a good idea to have something like this before #15584 is merged.

### CODE REVIEW

**Review by fanquake — APPROVED** [review]
ACK 1153caf78e21f6a9406430b704c7c74eb90975ca

Whilst I agree that some more spacing would be nice, if this commit doesn't get updated it can just be merged as is. Did bare-minimum testing on macOS.

message

**Comment by laanwj** [comment]
With so much text, it might make sense to turn this into an a popup instead of a notification. No one reads so much text in a notification so quickly.

**Comment by TheBlueMatt** [comment]
Maybe drop the last sentence or just rephrase to "Cannot process payment request - BIP70 has been disabled by default due to widespread security flaws. Ignore any recommendations to change wallets to use BIP 70".

File: src/qt/paymentserver.cpp [hunk 1]
```diff
@@ -328,7 +328,9 @@ void PaymentServer::handleURIOrFile(const QString& s)
 #ifndef ENABLE_BIP70
                     if (uri.hasQueryItem("r")) {  // payment request
                         Q_EMIT message(tr("URI handling"),
-                            tr("Cannot process payment request because BIP70 support was not compiled in."),
+                            tr("Cannot process payment request because BIP70 support was not compiled in.")+
+                            tr("Due to widespread security flaws in BIP70 it's strongly recommended that any merchant instructions to switch wallets be ignored.")+
```
[the remark this answers is not in the document]
Comment (jameshilliard) [reply src/qt/paymentserver.cpp]: I'll probably do a follow up commit to clean up formatting once I'm no longer traveling.

**Comment by jameshilliard** [comment]
@TheBlueMatt I think it's important to have an explicit recommendation on what's needed to fix the error.

**Comment by TheBlueMatt** [comment]
Hmm, but its ultimately not up to the user - asking for BIP21 from the merchant isn't so different from asking for "not BIP70".

**Comment by jameshilliard** [comment]
Well it's up to the merchant ultimately to support it, but if a bunch of users submit support requests to the merchant asking for BIP21 compatible URI's then it's much more likely the merchant will add support for BIP21 due to customer demand as opposed to potentially some other protocol.

**Comment by TheBlueMatt** [comment]
I mean BIP21 is really just an address and bitcoin: in front, I don't know that the response would be anything but that, no?

**Comment by jameshilliard** [comment]
Well you would want to specify BIP21 so that the merchant doesn't just suggest a [different incompatible BIP70-like payment method](https://github.com/bitpay/jsonPaymentProtocol).

**Comment by TheBlueMatt** [comment]
Does literally anyone implement that? I really don't think it's unclear and no users are gonna read that long text.

**Comment by jameshilliard** [comment]
> Does literally anyone implement that?

Bitpay does apparently as an alternative to BIP70.

**Comment by TheBlueMatt** [comment]
Lol I meant on the client side.

**Comment by jameshilliard** [comment]
Well I think the point is to ensure that customers are not just requesting a BIP70 alternative but BIP21 specifically as Bitpay is likely to suggest using jsonPaymentProtocol if the request is not explicitly for BIP21 support. I wonder if it would make sense to link to a tool like https://github.com/achow101/payment-proto-interface in the error message for decoding bitpay requests.

**Comment by TheBlueMatt** [comment]
I mean I'm not sure how you capture that in a way that (a) users understand and (b) is actually short enough that any user will read it (cause the current text no one is gonna read).

**Comment by jonasschnelli** [comment]
Going to merge this.
Turing into a popup, if that makes sense, can be done later.

**Pull request merged**